### PR TITLE
Resolved sorting issue of pairwise output files for author DE (SCP-5204)

### PR DESCRIPTION
**Pull request by michelle-vel**
Added correct natural sorting to the final file names generated from author DE. This is necessary to ensure that results can be displayed in the UI/UX, since it is sorting dependent. 

**Before**, showing sorting issue (note especially the file All_Cells_UMAP--General_Celltype--eosinophils--CSN1S1_macrophages--study--wilcoxon.tsv):
Screen Shot 2023-06-28 at 12 54 26 PM

**After:**
Screen Shot 2023-06-28 at 12 52 17 PM

### CODE REVIEW

File: ingest/author_de.py [hunk 1]
```diff
@@ -276,11 +282,12 @@ def check_group(names_dict, name):
 def sort_comparison(ls):
     """
     Naturally sort groups in a pairwise comparison; specially handle one-vs-rest
-    this should take in only a list of two, ex (type1 type2)
+    this should take in a combined name, such as B cells--CSN1S1 macrophages
```
Comment (eweitz) [inline ingest/author_de.py:285]: The usage reference passes in a list of groups, and the implementation here also operates on a list:

```suggestion
    this should take in a list of groups, such as ['B cells', 'CSN1S1 macrophages']
```

File: ingest/author_de.py [hunk 2]
```diff
@@ -81,10 +81,10 @@ def generate_result_files(self, col, genes, rest, groups, clean_val, qual):
...
         Then final format should have type 0 type 1 in the title, and genes, logfoldchanges, qval, and mean as columns
         """
-        # for i in clean_val:
-        #     val_to_sort = [i[0], i[1]]
-        #     sorted_list = sort_comparison(val_to_sort)
-        #     i[0], i[1] = sorted_list
+        #for i in clean_val:
+        #    val_to_sort = [i[0], i[1]]
+        #    sorted_list = sort_comparison(val_to_sort)
+        #   i[0], i[1] = sorted_list
```
Comment (eweitz) [inline ingest/author_de.py:87]: This seems removable.

```suggestion
```

**Review by eweitz — APPROVED** [review]
Thanks for the quick fix, and the effective PR description!

I suggest a non-blocking refinement for maintainability.

**Review by bistline — APPROVED** [review]

**Comment by eweitz** [comment]
I just appreciated the PR notes:

> Added correct lexical sorting

[The code in this PR](https://github.com/broadinstitute/scp-ingest-pipeline/pull/314/files#diff-bbdfb888de016bb1c7757e93a503e96cda31a52302c0173a73513ed9b79daa2cR282) actually adds correct _natural_ sorting, which [importantly differs](https://stackoverflow.com/a/6810681) from lexical sorting.  Lexical sorting (a.k.a. lexicographic sorting) is the default in most string sorting methods, but it unintuitively handles numbers in strings

In lexical sorting, "type_10" gets ordered before "type_2".  In natural sorting, "type_2" gets ordered before "type_10".

**Pull request merged**